### PR TITLE
Project: Associations - Private Events

### PR DESCRIPTION
**Pull request by SmithVyne**
This project was built as to be conceptually similar to 'eventbrite' with the aim to demonstrate the use of Rails associations especially for intertwined many to many relationships.
Associations between users - (creator and attendee) and events_attended and events_created. During this project, we accomplished numerous tasks that included:

- Creating a new user by signing up
- Allowing users to create events when logged in
- Allowing users to attend events
- Show past and upcoming events.

### CODE REVIEW

File: db/migrate/20200805013639_create_event_attendances.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateEventAttendances < ActiveRecord::Migration[6.0]
+  def change
+    create_table :event_attendances do |t|
+      t.references :attendee
+      t.references :attended_event
+
+      t.timestamps
```
Comment (maelfosso) [inline db/migrate/20200805013639_create_event_attendances.rb:7]: Add indexes for those two columns, please.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -7,9 +7,55 @@
...
+      <% end %>
+    <% end %>
+
+    <% if flash[:errors] %>
+      <% flash[:errors].each do |error| %>
+        <div class="alert alert-danger">
+          <%= error %>
+        </div>
+      <% end %>
+    <% end %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb:29]: `if`, `if/else` if what we called **logic in views**. It's not allowed to have logic in views. Kindly use **view helpers** for that.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -7,9 +7,55 @@
...
+          <% if current_user %>
+            <a class="nav-link active" href="/logout">Logout</a>
+            <a class="nav-link active" href="<%= user_path(current_user) %>">Dashboard</a>
+            <a class="nav-link active" href="<%= new_event_path %>">Create Event</a>
+          <% end %>
+          <a class="nav-link active" href="/events">All Events</a>
+          <a class="nav-link active" href="/login">Login</a>
+          <a class="nav-link active" href="/signup">Signup</a>
+        </div>
+      </div>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb:52]: Use view helpers for handling logic into views

File: app/views/events/_attendees.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <div class="card-header bg-info text-white">
+    Attendees
+  </div>
+  <% if event.attendees %>
+    <ul class="list-group list-group-flush">
+      <% event.attendees.each do |attendee| %>
+        <li class="list-group-item bg-warning"><%= attendee.name %></li>
+      <% end %>
+    </ul>
+  <% end %>
```
Comment (maelfosso) [inline app/views/events/_attendees.html.erb:11]: use view helpers

File: app/views/events/_events.html.erb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+              <li class="list-group-item bg-warning">
+                <%= event.title %> <br>
+
+                <a href="<%= event_path(event) %>" class="badge badge-light">Show Event</a>
+                <% if  user_attended_event?(session[:user_id], event.id) %>
+                  <span class="badge badge-light">You attended</span>
+                <% end %>
+                
+              </li>
+            <% end %>
```
Comment (maelfosso) [inline app/views/events/_events.html.erb:20]: Use view helpers

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (SmithVyne) [reply app/views/layouts/application.html.erb:52]: Hello @maelfosso, for large bodies of HTML like this, it is actually better to use conditions in the template, that's what they are meant for. Using view helpers is wrong for this, and highly unmaintainable.

**Pull request merged**